src/Reflection/ReflectionClass.php

```diff
@@ -627,13 +628,13 @@ public function getTraitAliases()
      *
      * @link http://php.net/manual/en/reflectionclass.getinterfaces.php
      *
-     * @param SourceLocator|null $sourceLocator a source locator - if none is provided, an autoloader-based locator
-     *                                          will be used
+     * @param SourceLocator $sourceLocator a source locator - if none is provided, an autoloader-based locator
+     *                                     will be used
      *
      * @return ReflectionClass[] An associative array of interfaces, with keys as interface names and the array
      *                           values as {@see ReflectionClass} objects.
      */
-    public function getInterfaces(SourceLocator $sourceLocator = null)
+    public function getInterfaces(SourceLocator $sourceLocator)
```

Comment:
This was changed because the previous logic didn't account for a null source locator. This should be handled again when we have an idea of what we want when we deal with the injected reflector.